hack/go_container.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/sh
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Simple posix sh reproducible go build container script with caching.
+#
+# Usage:
+#  hack/go_container.sh go version
+#  hack/go_container.sh go build -o /out/kind .
+set -o nounset -o errexit
+
+# ============================ SCRIPT SETTINGS =================================
+# get the repo root for defaulting OUT_DIR and SOURCE_DIR
+REPO_ROOT="${REPO_ROOT:-$(cd "$(dirname "$0")/.." && pwd)}"
+# output directory, will be mounted to /out, defaults to /bin in REPO_ROOT
+OUT_DIR="${OUT_DIR:-${REPO_ROOT}/bin}"
+# source directory, will be mounted to /src, defaults to current directory
+SOURCE_DIR="${SOURCE_DIR:-$(pwd -P)}"
+# default to disabling CGO for easier reproducible builds and cross compilation
+export CGO_ENABLED="${CGO_ENABLED:-0}"
+# the container image, by default a recent official golang image
```

Comment:
when do you say recent do you want to do this smarter?
https://gist.github.com/n8henrie/1043443463a4a511acf98aaa4f8f0f69#file-get_golang-sh-L11

Reply:
we could use `golang:latest` but that might get a bit surprising ... probably best to periodically manually update it.

Reply:
(also `golang:latest` actually works currently, but it's not reproducible...)